src/uri/validator/urivalidator.rs

```diff
@@ -186,19 +185,218 @@ impl UriValidator {
+    /// #                              0x03c, 0x03d, 0x03e, 0x03f, 0x040, 0x041, 0x042, 0x043, 0x044, 0x045,
+    /// #                              0x046, 0x047, 0x048, 0x049, 0x04a, 0x04b, 0x04c, 0x04d, 0x04e, 0x04f,
+    /// #                              0x050, 0x051, 0x052, 0x053, 0x054, 0x055, 0x056, 0x057, 0x058, 0x059,
+    /// #                              0x05a, 0x05b, 0x05c, 0x05d, 0x05e, 0x05f, 0x060, 0x061, 0x062, 0x063,
+    /// #                              0x064, 0x065, 0x066, 0x067, 0x068, 0x069, 0x06a, 0x06b, 0x06c, 0x06d,
+    /// #                              0x06e, 0x06f, 0x070, 0x071, 0x072, 0x073, 0x074, 0x075, 0x076, 0x077,
+    /// #                              0x078, 0x079, 0x07a, 0x07b, 0x07c, 0x07d, 0x07e, 0x07f, 0x080, 0x081,
+    /// #                              0x082, 0x083, 0x084, 0x085, 0x086, 0x087, 0x088, 0x089, 0x08a, 0x08b,
+    /// #                              0x08c, 0x08d, 0x08e, 0x08f, 0x090, 0x091, 0x092, 0x093, 0x094, 0x095,
+    /// #                              0x096, 0x097, 0x098, 0x099, 0x09a, 0x09b, 0x09c, 0x09d, 0x09e, 0x09f,
+    /// #                              0x0a0, 0x0a1, 0x0a2, 0x0a3, 0x0a4, 0x0a5, 0x0a6, 0x0a7, 0x0a8, 0x0a9,
+    /// #                              0x0aa, 0x0ab, 0x0ac, 0x0ad, 0x0ae, 0x0af, 0x0b0, 0x0b1, 0x0b2, 0x0b3,
+    /// #                              0x0b4, 0x0b5, 0x0b6, 0x0b7, 0x0b8, 0x0b9, 0x0ba, 0x0bb, 0x0bc, 0x0bd,
+    /// #                              0x0be, 0x0bf, 0x0c0, 0x0c1, 0x0c2, 0x0c3, 0x0c4, 0x0c5, 0x0c6, 0x0c7,
+    /// #                              0x0c8, 0x0c9, 0x0ca, 0x0cb, 0x0cc, 0x0cd, 0x0ce, 0x0cf, 0x0d0, 0x0d1,
+    /// #                              0x0d2, 0x0d3, 0x0d4, 0x0d5, 0x0d6, 0x0d7, 0x0d8, 0x0d9, 0x0da, 0x0db,
+    /// #                              0x0dc, 0x0dd, 0x0de, 0x0df, 0x0e0, 0x0e1, 0x0e2, 0x0e3, 0x0e4, 0x0e5,
+    /// #                              0x0e6, 0x0e7, 0x0e8, 0x0e9, 0x0ea, 0x0eb, 0x0ec, 0x0ed, 0x0ee, 0x0ef,
+    /// #                              0x0f0, 0x0f1, 0x0f2, 0x0f3, 0x0f4, 0x0f5, 0x0f6, 0x0f7, 0x0f8, 0x0f9,
+    /// #                              0x0fa, 0x0fb, 0x0fc, 0x0fd, 0x0fe, 0x0ff, 0x000];
+    ///
+    /// let uri = UUri {
+    ///     authority: Some(UAuthority {
+    ///         id: Some((0..=256) // <- note that ID will exceed 255 byte limit
+    ///                 .map(|i| (i % 256) as u8)
+    ///                 .collect::<Vec<u8>>()),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     entity: Some(UEntity {
+    ///         id: Some(29999),
+    ///         version_major: Some(254),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     resource: Some(UResource {
+    ///         id: Some(29999),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     ..Default::default()
+    /// };
+    /// # assert_eq!(uri.clone().authority.unwrap().id.unwrap(), expected_id_bytes);
```

Comment:
what is the point of this and having the expected_id_bytes?

Reply:
Oh yeah, this is the bit from the rustdoc book about [hiding parts of your doctest](https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example) which may not be relevant or they key thing you wanna showcase.

Reply:
OK -- I'll remove this, sorry I misread! :slightly_smiling_face: